extension/background/pageMetadata-contentScript.js

```diff
@@ -41,5 +41,105 @@ this.pageMetadataContentScript = (function() {
     true
   );
 
+  function cleanURL(details) {
+    const globalBlockedParams = [
+      "utm_source",
+      "utm_medium",
+      "utm_campaign",
+      "utm_term",
+      "utm_content",
+    ];
+
+    function getParams(URL) {
```

Comment:
Instead of all this string-based splitting you can use the `URL` object, kind of like:

```js
const u = new URL(url);
const params = u.searchParams;
params.delete("utm_source");
const betterUrl = u.toString();
```

Obviously doing the other cleanup too. You can read about searchParams here: https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams

The cool thing about URL objects is you can edit them and then turn them back into strings.